addons/wowl/static/src/components/loading_indicator/loading_indicator.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, useState } from "@odoo/owl";
+import type { OdooEnv } from "../../types";
+
+export class LoadingIndicator extends Component<{}, OdooEnv> {
+  static template = "wowl.LoadingIndicator";
+  state = useState({
+    count: 0,
+    isShown: false,
+  });
+  rpcIds: Set<number> = new Set();
+  blockUITimer!: number;
+
+  constructor() {
+    super(...arguments);
+    this.env.bus.on("RPC:REQUEST", this, (rpcId: number) => {
+      this.requestCall(rpcId);
+    });
+    this.env.bus.on("RPC:RESPONSE", this, (rpcId: number) => {
+      this.responseCall(rpcId);
+    });
+  }
+
+  requestCall(rpcId: number) {
+    if (this.state.count === 0) {
+      this.state.isShown = true;
+      this.blockUITimer = setTimeout(() => {
+        this.env.services.ui.block();
+      }, 3000);
```

Comment:
This would not work ?
```suggestion
      this.blockUITimer = setTimeout(this.env.services.ui.block, 3000);
```